### ISSUE TITLE
Adding Features During Runtime

### ISSUE THREAD

**Issue by SoulOfSet** [mapbox/mapbox-unity-sdk#1542]
I'm attempting to add prefabs to a MapBox map instance dynamically using data I'm getting from a custom source, not from the MapBox Studio API.

I've been looking at the SpawnPrefabAtGeoLocation method, and I think I could see how I could edit the coordinates in the PrefabItemOptions object contained within a prefab layer to get what I need done, but I think that this would be more appropriately done as a feature rather than a POI. 

I've been digging through the vector visualizer logic for the features but I don't think I'm understanding where the features are actually defined and how I would build out new ones. Is there any functions built into the API that allow for this or is it something I will need to build out manually?  If so, could you give me some directions on what I could do to carry out this task. I think really that I am misunderstanding the general flow of the feature visualizer.

**Comment by brnkhy** [comment]
@SoulOfSet I'm not sure if I understand it correct; are you trying to spawn objects on map or create new layers in abstract map script?
I mean are you looking for converting latlng to map space?

**Comment by SoulOfSet** [comment]
Currently, I'm getting in coordinates from a separate source. At the moment, I'm just placing them on the terrain manually using GeoToWorldPosition and querying the height. 

I was wondering if instead if I could feed them into the abstract map class as a features instead though. Placing them manually on the map seems to have issues when dealing with maps that aren't worldscale or have terrain. It would be nice to also be able to utilize the modifiers on my custom features. 

Basically, I want to use the custom features section of the AbstractMap.cs without getting them in from MapBox, but instead adding them in manually after the tile has loaded.

**Comment by dm67x** [comment]
I don't know if it will work but try to update ReplaceFeatureModifier and then reload the map. I know that when we update a filter and reload the map it take into consideration what i have changed.

Hope you can solve your problem.

**Comment by brnkhy** [comment]
@SoulOfSet ah I'm afraid there's no way to feed external data to Abstract map like that. and it might be a little difficult to set it up but I believe everything you would need to create and use them externally, must be provided by abstract map (relative world scale etc).
What I generally do is to create a basic clone of modifier stack (or maybe existing stack itself) in a separate script, fill it with modifiers I want. I believe Directions example also does something similar.

**Comment by SoulOfSet** [comment]
Ah I see. I was looking into how geometries were parsed and that makes quite a bit of sense. Thanks for your help.
